samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -54,18 +52,31 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
       case _ => None
     }
 
+    val containerListener = new SamzaContainerListener {
+      override def onContainerFailed(t: Throwable): Unit = {
+        throw t
```

Comment:
Will the thrown exception be handled correctly or result in an uncaught exception and jvm quitting?